nco/custom.py

```diff
@@ -0,0 +1,268 @@
+import os
+import re
+import sys
+import numpy as np
+
+DEBUG=1
+
+# check mode
+validModes = dict({
+    'append': 'a',
+    'create': 'c',
+    'delete': 'd',
+    'modify': 'm',
+    'nappend': 'n',
+    'overwrite': 'o'
+})
+
+# netCDF type names
+netTypes = dict({
+    'NC_FLOAT': 'f',
+    'NC_DOUBLE': 'd',
+    'NC_INT': 'i',
+    'NC_SHORT': 's',
+    'NC_CHAR': 'c',
+    'NC_BYTE': 'b',
+    'NC_UBYTE': 'ub',
+    'NC_USHORT': 'us',
+    'NC_UINT': 'ui',
+    'NC_INT64': 'll',
+    'NC_UINT64': 'ull',
+    'NC_STRING': 'sng'
+})
+
+# numpy type names
+npTypes = dict({
+    'float32': 'f',
+    'float64': 'd',
+    'int32': 'i',
+    'int16': 's',
+    'char': 'c',
+    'byte': 'b',
+    'ubyte': 'ub',
+    'int8' : 'b',
+    'uint8' : 'ub',
+    'uint16': 'us',
+    'uint32': 'ui',
+    'int64': 'll',
+    'uint64': 'ull',
+    'string': 'sng'
+})
+
+npTypesNP = dict({
+    'float32' : np.float32, 
+    'float'   : np.float32 ,
+    'f'       : np.float32 ,
+    'float64' : np.float64,
+    'double'  : np.float64,
+    'd'       : np.float64,
+    'int32'   : np.int32,
+    'i'       : np.int32,
+    'l'       : np.int32,
+    'int16'   : np.int16,
+    's'       : np.int16,
+    'str'     : str,
+    'char'    : str,
+    'c'       : str,
+    'string'  : str,  
+    'sng'     : str,  
+    'byte'    : np.byte,
+    'b'       : np.byte,
+    'ubyte'   : np.ubyte,
+    'ub'      : np.ubyte,
+    'int8'    : np.byte,
+    'uint8'   : np.ubyte,
+    'uint16'  : np.uint16,
+    'us'      : np.uint16,
+    'uint32'  : np.uint32,
+    'u'       : np.uint32,
+    'ui'      : np.uint32,
+    'ul'      : np.uint32,
+    'int64'   : np.int64,
+    'll'      : np.int64,
+    'uint64'  : np.uint64,
+    'ull'     : np.uint64
+})
+
+
+
+__prog__ = os.path.splitext(os.path.basename(__file__))[0]
+
+
+class atted(object):
+
+    def __init__(self, mode='overwrite',attName="",varName="", Value=None,  sType=None,**kwargs):
+    #def __init__(self, **kwargs):
+        mode = kwargs.pop('mode', mode)
+        attName = kwargs.pop('attName', attName)
+        varName = kwargs.pop('varName', varName)
+        Value = kwargs.pop('Value', Value)
+        sType = kwargs.pop('sType',sType )
+            
+        if mode in validModes:
+           mode=validModes[mode]  
+        elif mode in validModes.keys():
+           pass
+        else:
+           raise Exception('{0}: mode "{1}" not found'.format("atted()","atted():",mode))
+        
+        if attName==None or len(attName)==0: 
+            raise Exception ('{0}: attName zero length\n'.format("atted()"))
```

Comment:
the proper way to compare to `None` uses `is`:

```Python
if att_name is None or len(att_name) == 0: 
```

This could also be written as:
```Python
if not att_name:
    raise ValueError('att_name is required'"))
```